pkg/tfbridge/provider_test.go

```diff
@@ -799,6 +799,54 @@ func TestProviderPreviewV2(t *testing.T) {
 	}).DeepEquals(outs["nestedResources"]))
 }
 
+func TestProviderCheckWithAutonaming(t *testing.T) {
+	t.Parallel()
+	provider := &Provider{
+		tf:     shimv1.NewProvider(testTFProvider),
```

Comment:
This uses the TF SDKv1 - that is long deprecated and very few of our providers use it. Can you instead replace `shimv1` with `shimv2` here to use the TF SDKv2 for these tests?